typescript/infra/scripts/deploy-igp.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  AbacusCore,
+  ChainMap,
+  ChainName,
+  CoreConfig,
+  CoreContracts,
+  MultiProvider,
+  objMap,
+  serializeContracts,
+} from '@abacus-network/sdk';
+
+import { AbacusCoreInfraDeployer } from '../src/core/deploy';
+import { readJSON, writeJSON } from '../src/utils/utils';
+
+import {
+  getCoreContractsSdkFilepath,
+  getCoreEnvironmentConfig,
+  getCoreVerificationDirectory,
+  getEnvironment,
+} from './utils';
+
+class IGPDeployer<
+  Chain extends ChainName,
+> extends AbacusCoreInfraDeployer<Chain> {
+  constructor(
+    multiProvider: MultiProvider<Chain>,
+    configMap: ChainMap<Chain, CoreConfig>,
+    public readonly existingContracts: ChainMap<
+      Chain,
+      CoreContracts<Chain, Chain>
+    >,
+  ) {
+    super(multiProvider, configMap);
+  }
+
+  async deployIGP() {
+    const chains = this.multiProvider.chains();
+    for (const chain of chains) {
+      const contracts = this.existingContracts[chain];
+      const interchainGasPaymaster = await this.deployProxiedContract(
+        chain,
+        'interchainGasPaymaster',
+        [],
+        contracts.upgradeBeaconController.address,
+        [],
+      );
+      contracts.interchainGasPaymaster = interchainGasPaymaster;
```

Comment:
the indirection with `contracts` is hard to read
consider directly assigning to `this.existingContracts[chain].interchainGasPaymaster`

Reply:
or use `objMap`

Reply:
Liked the former suggestion